full/coupler_main.F90

```diff
@@ -464,7 +466,8 @@ program coupler_main
   integer                              :: num_ice_bc_restart=0, num_ocn_bc_restart=0
   type(time_type)                      :: Time_restart, Time_restart_current, Time_start
   character(len=32)                    :: timestamp
-
+  type(FmsNetcdfDomainFile_t)          :: ice_restart_fileobj !< restart file object returned by call
+                                                              !! open_file
```

Comment:
Why isn't this done in the ice model like all of the other restarts?

Reply:
It is;the ice restart fileobject is opened and registered to in ice_model_init, written out in ice_model_restart, and closed in ice_model_end, all of which are called by the coupler. Passing the fileobject around preserves the current interface design, and avoids having to reopen the file and re-register the restart variables before writing. I could make the file object a module variable in ice_type, but I'm not sure if this would cause problems for cases when intermediate restarts are written, which is handled by coupler_main.

Reply:
None of the other components have their restart files in the coupler, so I guess I'm just not understanding what is different about the ice.